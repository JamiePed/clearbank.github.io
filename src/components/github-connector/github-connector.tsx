import React from 'react'
import Popover from 'element-react/dist/npm/es6/src/popover'

import * as Types from './github-connector.types'
import * as Styles from './github-connector.styles'

const GithubConnector: React.FunctionComponent<Types.GithubConnectorProps> = ({
  filePath
}): JSX.Element => (
  <Popover
    placement='left'
    width='225'
    trigger='hover'
    content='Edit this section on GitHub'
  >
    <Styles.ShareLink href={filePath} target='_blank'>
      <Styles.Icon className='cb-icon-edit' />
    </Styles.ShareLink>
  </Popover>
)

export default GithubConnector
