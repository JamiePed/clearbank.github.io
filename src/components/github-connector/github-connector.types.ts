export interface GithubConnectorProps {
  children?: React.ReactNode
  filePath: string
}
