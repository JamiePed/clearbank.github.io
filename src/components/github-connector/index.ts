export { default } from './github-connector'
