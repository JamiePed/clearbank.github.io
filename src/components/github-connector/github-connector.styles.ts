import styled from 'styled-components'

import { colors } from 'src/components/theme'

import {
  animationButtonIn,
  animationButtonOut
} from 'src/components/clearbank-ui/styles'

const transitionDuration = '0.5s'

// todo: replace with new button design from new styleguide once available
export const ShareLink = styled.a`
  display: block;
  height: 40px;
  width: 40px;
  background-color: ${colors.brandGrayLightest};
  color: ${colors.businessBlue};
  border-radius: 50%;
  transition: all ${transitionDuration};
  overflow: hidden;

  &:hover {
    background: ${colors.brandSecondaryDarkest};
    color: ${colors.brandLight};
  }

  &:focus,
  &:active {
    background: ${colors.brandInfoDark};
    color: ${colors.brandLight};
  }
`

export const Icon = styled.i`
  display: block;
  padding: 10px;
  font-size: 20px;

  // hover on parent
  ${ShareLink}:hover & {
    animation: ${animationButtonIn} 0.15s ease-in,
      ${animationButtonOut} 0.15s 0.15s ease-out;
  }
`
