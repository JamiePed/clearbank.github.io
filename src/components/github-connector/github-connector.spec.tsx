import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './github-connector'

import * as Types from './github-connector.types'

// Mock popover element
jest.mock('element-react/dist/npm/es6/src/popover', () => {
  const Popover = props => <div className='popover-element'>{props.children}</div>

  return Popover
})

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.GithubConnectorProps = {
  filePath: 'file/path.mdx'
}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })

  test('has child elements', () => {
    expect(root.hasChildNodes()).toBeTruthy()
  })

  test('has filePath attribute from props', () => {
    expect(root.querySelector('a')).toHaveAttribute('href')
    expect(root.querySelectorAll(`[href="${defaultProps.filePath}"]`).length)
      .toBeTruthy
  })

  test('has target _blank attribute', () => {
    expect(root.querySelector('a')).toHaveAttribute('target')
    expect(root.querySelectorAll('[target="_blank"]').length).toBeTruthy
  })

  test('has correct icon', () => {
    expect(root.querySelectorAll('.cb-icon-edit').length).toBeTruthy
    expect(root.querySelector('.cb-icon-edit').tagName.toLowerCase()).toBe('i')
  })
})
