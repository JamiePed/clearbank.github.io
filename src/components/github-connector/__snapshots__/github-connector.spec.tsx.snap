// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
.c1 {
  display: block;
  height: 40px;
  width: 40px;
  background-color: #F3F8FB;
  color: #031563;
  border-radius: 50%;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  overflow: hidden;
}

.c1:hover {
  background: #0D326E;
  color: #fff;
}

.c1:focus,
.c1:active {
  background: #0B2145;
  color: #fff;
}

.c2 {
  display: block;
  padding: 10px;
  font-size: 20px;
}

.c0:hover .c2 {
  -webkit-animation: frYDOQ 0.15s ease-in, kjJcRz 0.15s 0.15s ease-out;
  animation: frYDOQ 0.15s ease-in, kjJcRz 0.15s 0.15s ease-out;
}

<div
  class="popover-element"
>
  <a
    class="c0 c1"
    href="file/path.mdx"
    target="_blank"
  >
    <i
      class="c2 cb-icon-edit"
    />
  </a>
</div>
`;
